tests/jobs/test_local_task_job.py

```diff
@@ -406,3 +409,23 @@ def success_callback(context):
         self.assertTrue(data['called'])
         process.join(timeout=10)
         self.assertFalse(process.is_alive())
+
+
+class TestLocalTaskJobPerformance:
+    @pytest.mark.parametrize("side_effects", [[0], 9 * [None] + [0]])  # type: ignore
+    @mock.patch("airflow.jobs.local_task_job.get_task_runner")
+    def test_number_of_queries_single_loop(self, mock_get_task_runner, side_effects):
+        unique_prefix = str(uuid.uuid4())
+        dag = DAG(dag_id=f'{unique_prefix}_test_number_of_queries', start_date=DEFAULT_DATE)
+        task = DummyOperator(task_id='test_state_succeeded1', dag=dag)
+
+        dag.clear()
+        dag.create_dagrun(run_id=unique_prefix, state=State.NONE)
```

Comment:
Can you remove dagrun in teardown to avoid side-effect?

Reply:
@mik-laj done